File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int tGas=0,tCost=0,cGas=0;
        int start=0;
        for(int i=0;i<gas.size();i++){
            tGas+=gas[i];
            tCost+=cost[i];
            cGas +=gas[i]-cost[i];
            if(cGas<0){
                cGas=0;
                start=i+1;
            }
        }
        if(tGas < tCost) return -1;
        return start;
    }
};

File: 0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalTank =0, curTank =0, start =0;
        for(int i=0;i<gas.length;i++){
            totalTank += gas[i] - cost[i];
            curTank += gas[i] - cost[i];
            if(curTank < 0){
                curTank =0;
                start = i+1;
            }
        }
        return totalTank < 0? -1: start;
    }
}